clustering/ejb/WFLY-7628_Distributed_EJB_Timers.adoc

```diff
@@ -0,0 +1,142 @@
+** This has the advantage of supporting passivation of timer instances to support applications with a sufficiently large number of transient timers in excess of some configured maximum.
+* Add support to TimerServiceMetaData/TimerServiceMetaDataParser for overriding the persistent timer manager per EJB.
+
+=== Non-Requirements
+
+== Backwards Compatibility
+
+* Servers started using an old version of the EJB subsystem configuration will continue to behave as normal, using the existing TimerServiceImpl for both transient and persistent timers.
+
+=== Default Configuration
+
+The following caches will be added to the default ejb cache-container configuration with the Infinispan subsystem:
+
+For the HA profiles:
+
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent-timers:add()
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent-timers/component=locking:add(isolation=PESSIMISTIC)
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent-timers/component=transaction:add(mode=BATCH)
+	/subsystem=infinispan/cache-container=ejb/distributed-cache=persistent-timers/store=file:add(purge=false)
+
+And for non-HA profiles:
+
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent-timers:add()
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent-timers:component=locking:add(isolation=PESSIMISTIC)
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent-timers:component=transaction:add(mode=BATCH)
+	/subsystem=infinispan/cache-container=ejb/local-cache=persistent-timers/store=file:add(purge=false,passivation=false)
+
+The following timer management provider will be added to the default distributable-ejb subsystem configuration:
+
+	/subsystem=distributable-ejb/infinispan-timer-management=persistent:add(cache-container=ejb, cache=persistent-timers)
+
+The default timer-service configuration of the EJB subsystem will be modified as follows:
+
+	/subsystem=ejb3/service=timer-service:undefine-attribute(name=default-data-store)
+	/subsystem=ejb3/service=timer-service:write-attribute(name=persistent-timer-management, value=persistent-timers)
+
+=== Importing Existing Configuration
+
+=== Deployments
+
+=== Interoperability
+
+== Implementation Plan
+
+The feature requires WFLY-14953 in order to house the configuration for distributed timer management providers.
+
+== Security Considerations
+
+////
+Identification if any security implications that may need to be considered with this feature
+or a confirmation that there are no security implications to consider.
+////
+
+== Test Plan
+
+* All existing TimerService tests must pass against the new persistent timer implementation, including the EJB TCK.
+* Existing distributed EJB timer tests that use the JDBC timer-store within an EAP cluster may need to be modified since timers may execute on different cluster members than expected by the tests.
```

Comment:
I did not change the existing test (there is only 1 which only runs against a single server), but rather added a separate integration test to the clustering testsuite that validates timer failover.

Reply:
Ok, thanks. Could please add this details to the document itself? Also, the involved testsuite module should be mentioned (better linked) here, since it is not implied - nor required - to look at the PR for understanding what tests will be affected. Indeed, it should be made visible here.

Reply:
I'm talking about the standard Jakarta EE TCK: https://github.com/eclipse-ee4j/jakartaee-tck

I also misspoke, the TCK is not composed of any modules, but rather one can run a set of tests within a given directory (as with any junit testsuite).

Reply:
No existing tests will be affected, but I'll better describe the test scenarios being added to the clustering integration testsuite.